### PR TITLE
Many improvements in class MacAddress

### PR DESCRIPTION
**Pull request by gx740**
- The implementation of many constructors and methods have been moved into header file, the implementation of method `operator==` is simplified and optimized.

- Two errors in method `init()` were fixed:
  1. The buffer overflow occures when the length of input argument is more than 6.
  2. For incorrect MAC address string that contains the wrong hexadecimal number the old version of this method is setting `m_IsValid` to true value.

- The qualifier `const` was added to many methods

- The operator of assignment and the copy constructor are unnecessary.

- The new constructor `MacAddress(std::initializer_list<uint8_t>)` was added for the compilers which are supporting C++11 language standard.

- Operator `'sizeof m_Address'` is now used in all cases instead of number 6 to avoid the potential mistakes with buffer overflow.

- Tests were added.

### CODE REVIEW

File: Tests/Packet++Test/main.cpp [hunk 1]
```diff
@@ -7113,6 +7113,50 @@ PACKETPP_TEST(RadiusLayerEditTest)
 	PACKETPP_TEST_PASSED;
 }
 
+PACKETPP_TEST(MacAddressTests)
```
Comment (seladb) [inline Tests/Packet++Test/main.cpp]: There is already existing UT for `MacAddress` in Pcap++Test: https://github.com/seladb/PcapPlusPlus/blob/7bc25adc8f5e2c0dc59710a3fa94b6956498a298/Tests/Pcap%2B%2BTest/main.cpp#L663

Can you move these UTs to there?

File: Common++/header/MacAddress.h [hunk 2]
```diff
@@ -52,61 +57,70 @@ namespace pcpp
...
 		 */
-		std::string toString();
+		std::string toString() const;
 
 		/**
 		 * Allocates a byte array of length 6 and copies address value into it. Array deallocation is user responsibility
 		 * @param[in] arr A pointer to where array will be allocated
 		 */
-		void copyTo(uint8_t** arr);
+		void copyTo(uint8_t** arr) const;
```
Comment (seladb) [inline Common++/header/MacAddress.h]: If moving most of the method implementations to the header file, why not moving the implementation of this method as well?

Comment (gx740) [reply Common++/header/MacAddress.h]: I'll do it

File: Tests/Packet++Test/main.cpp (hunk 1 above)
Comment (gx740) [reply Tests/Packet++Test/main.cpp]: Ok

File: Common++/header/MacAddress.h (hunk 2 above)
Comment (gx740) [reply Common++/header/MacAddress.h]: Done

File: Tests/Packet++Test/main.cpp (hunk 1 above)
Comment (gx740) [reply Tests/Packet++Test/main.cpp]: Done

**Pull request merged**